src/sdk/service/network-validation-service/network-validation-service.ts

```diff
@@ -0,0 +1,475 @@
+  }
+
+  /**
+   * Gets a string representation of an IP address range without the .0 and .255 if applicable. Returns null if either
+   * the netmask or ipAddress are invalid.
+   * @param {string} netmask the netmask string
+   * @param {string} ipAddress the IP address string
+   * @return {string} a string representation of the IP address range. For
+   *         example, "192.168.0.1 - 192.168.0.10"
+   */
+  getSubnetRangeStrWithoutReserved(netmask: string, ipAddress: string): string | null {
+    const range = this.getSubnetRange(netmask, ipAddress);
+    if (!range) {
+      return null;
+    }
+    const modRange = {
+      start: this.getIntegerAsIp(this.getIpAsInteger(range.start) + 1),
+      end: this.getIntegerAsIp(this.getIpAsInteger(range.end) + 1)
+    };
+    return modRange.start + '-' + modRange.end;
+  }
+
+  /**
+   * Indicates whether an IP address belongs to a subnet. Returns null if any of
+   * the parameters do not represent valid network identifiers.
+   * @param {string} netmask the netmask of the subnet
+   * @param {string} netIpAddress any IP address on the subnet
+   * @param {string} testIpAddress the IP address which should be tested for
+   *          whether it belongs to the subnet
+   * @return {boolean} boolean value
+   */
+  isInSubnet(netmask: string, netIpAddress: string, testIpAddress: string): boolean | null {
+    if (!(this.isValid(netIpAddress) && this.isValidNetmask(netmask) && this.isValid(testIpAddress))) {
+      return null;
+    }
+    const netmaskInt = this.getIpAsInteger(netmask);
+    const netIpInt = this.getIpAsInteger(netIpAddress);
+    const testIpInt = this.getIpAsInteger(testIpAddress);
+    return ((netmaskInt & netIpInt) === (netmaskInt & testIpInt));
+  }
+
+  /**
+   * Indicates whether an IP address is in a range of IP addresses. Returns null
+   * if any of the parameters do not represent valid network identifiers.
+   * @param {string} ipAddress the IP address which should be tested for whether
+   *          it is in the range
+   * @param {object} range an object of the form { start: "x.x.x.x", end:
+   *          "x.x.x.x" }, representing the range of IP addresses.
+   * @return {boolean} boolean value
+   */
+  isInRange(ipAddress: string, range: Range): boolean {
+    if (this.isValid(ipAddress) && this.isValid(range.start) && this.isValid(range.end)) {
+      const testComponents = ipAddress.split('.');
+      const lowComponents = range.start.split('.');
+      const highComponents = range.end.split('.');
+      for (let i = 0; i < 4; i++) {
+        if (!this.inRange(lowComponents[i], highComponents[i], testComponents[i])) {
+          return false;
+        }
+      }
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Takes a set of IP addresses and a set of IP ranges, and indicates whether
+   * any of the IP addresses belong to any of the ranges.
+   * @param {array} ips of IP strings
+   * @param {array} ranges of range objects
```

Comment:
Typescript way of typing arrays would be Array<string> or string[] just like how they're specified in the function parameters.